src/jit/namedintrinsiclist.h

```diff
@@ -66,6 +66,20 @@ enum HWIntrinsicFlag : unsigned int
     // Copy Upper bits
     // some SIMD scalar intrinsics need the semantics of copying upper bits from the source operand
     HW_Flag_CopyUpperBits = 0x200,
+
+    // Invert integral operands
+    HW_Flag_InvertIntegralOperands = 0x400,
```

Comment:
Why do we need "Integral" in the name? Will you add a `HW_Flag_InvertFloatOperands`?

BTW, I do not think we need this flag. This intrinsic can be `HW_Category_Helper` with `HW_Flag_NoCodeGen`.

Reply:
Looks like you added three new flags to handle one intrinsic `CompareLessThan` (any other?) that even influence other intrinsic definitions.

It is not worthwhile to change the frame just for one or few intrinsic. Giving `HW_Categoty_Special` to `CompareLessThan` can address all the problems.

Reply:
> What do you mean by invert operands, and why can it not be done for double or float?

That is unfortunate name as I see it now as for. Much better name would be `HW_Flag_ReverseIntegralOperandsOrder` or similar but shorter. Some operations require passing operands in reverse order i.e. instead of `(op1, op2)` you have to pass `(op2, op1)`.

> Looks like you added three new flags to handle one intrinsic CompareLessThan

No, it  is not true, if you check PR code for all types where `HW_Flag_Imm...` is added in Sse2 ISA it is required in many more cases. It allows to keep at least 4 instructions in the table and actually makes additional type check on fixed immediate instructions what makes code more robust. Essentially majority of Sse2 scalar instructions would need `HW_Flag_Imm_Double` and there are quite a lot of them.

Reply:
> That is unfortunate name as I see it now as for. Much better name would be HW_Flag_ReverseIntegralOperandsOrder or similar but shorter. Some operations require passing operands in reverse order i.e. instead of (op1, op2) you have to pass (op2, op1).

Okay, but why can't the same be applied to float/double?

For float/double, there is actually no compare GT or compare GTE operations (on older hardware), and we have to explicitly call the inverse operation (compare LTE or compare LT respectively) and swap the operands as well.

The same logic should apply to both and will be useful for handling containment as well (since only one operand can be contained, we may want to swap the operands and the operation if op1 is containable but op2 isn't).

Reply:
> Okay, but why can't the same be applied to float/double?

> For float/double, there is actually no compare GT or compare GTE operations (on older hardware), and we have to explicitly call the inverse operation (compare LTE or compare LT respectively) and swap the operands as well.

I understand the point but for this particular intrinsic double in Sse2 and float in Sse are handled by `cmppd` and `cmpps` instructions respectively which do not require operand reversing but need fixed immediate value. So I have had to  differentiate between integral types and float types in this particular case.

Perhaps it would be useful to have more general `ReverseOperands` flag for the cases you have indicated or an additional `ReverseFloatOperands` flag just for floats.